backend/src/services/DuplicateDetectionService.ts

```diff
@@ -0,0 +1,52 @@
+import { EventRepository } from './EventRepository';
+
+export class DuplicateDetectionService {
+  private eventRepository: EventRepository;
+
+  constructor() {
+    this.eventRepository = new EventRepository();
+  }
+
+  /**
+   * Checks if an event is a duplicate based on title, date, and location
+   * @param title The title of the event
+   * @param date The date of the event
+   * @param location The location of the event
+   * @returns Promise<boolean> True if duplicate exists, false otherwise
+   */
+  async isDuplicate(title: string, date: Date, location: string): Promise<boolean> {
+    // Check if an event with the same title, date and location already exists
+    const existingEvent = await this.eventRepository.findByTitleDateLocation(title, date, location);
+
```

Comment:
this should only check title and maybe end date. location is too uncommon of a field to rely on